synapse/cores/common.py

```diff
@@ -1185,17 +1194,22 @@ def _getStormCore(self, name=None):
         return self
 
     def _check_axonclient(self):
-        if self.axon_client is None:
-            raise s_common.NoSuchOpt(name='axon:conf', mesg='The cortex does not have an axon configured')
+        if not self.cellpool_ready:
+            raise s_common.NoSuchOpt(name='cellpool:conf', mesg='The cortex does not have a cell pool configured')
+        if not self.axon_ready:
+            raise s_common.NoSuchOpt(name='axon:name', mesg='The cortex does not have an axon configured')
 
-    def _axonclient_wants(self, hashes):
-        # takes a list of sha256 hashes
+    def _get_axon_client(self):
         self._check_axonclient()
-        return self.axon_client.wants(hashes)
+        return s_axon.AxonClient(self.cellpool.get(self.axon_name))
```

Comment:
Not sure if I like this. Maybe I should just set self.axon_client? I need to test what happens if a client's session is finid